quickevent/app/quickevent/plugins/Event/src/services/oresultsclient.cpp

```diff
@@ -128,8 +128,9 @@ void OResultsClient::sendFile(QString name, QString request_path, QString file,
 	QUrl url(API_URL + request_path);
 	QNetworkRequest request(url);
 	QNetworkReply *reply = m_networkManager->post(request, multi_part);
+	multi_part->setParent(reply);
 
-	connect(reply, &QNetworkReply::finished, reply, [reply, name, on_success]()
+	connect(reply, &QNetworkReply::finished, this, [=]()
```

Comment:
It seems like attempt to solve importer crash. I do not think, that this will help. Original version was not leak, IMO. Passing this as context object doesn't do much, since o-result service object lives as long as application itself.

Reply:
the leak was referring to the `multi_part` object not having parent set and therefore never scheduled to be deleted.  

i switched `reply` to `this` for the context parameter, because i have never seen in other code, that the first parameter would also be used in the 3rd context parameter.

Reply:
Now I understand, so `multi_part->setParent(reply);` fixes memory leak, context object change is not related.

Context object is used to discard connection + lambda, when context object is deleted. If `reply` is signal sender, then context object is not needed indeed. But, `on_success` might use `this` in implementation, so assigning `this` to context-object might help, even if it is not clearly visible from code.